pkg/smb/nodeserver_test.go

```diff
@@ -277,135 +320,161 @@ func TestNodePublishVolume(t *testing.T) {
+	if err != nil {
+		t.Fatalf(fmt.Sprintf("failed to get fake mounter: %v", err))
 	}
+	d.mounter = mounter
 
 	for _, test := range tests {
-		_, err := d.NodePublishVolume(context.Background(), &test.req)
-		if !reflect.DeepEqual(err, test.expectedErr) {
-			t.Errorf("test case: %s, Unexpected error: %v", test.desc, err)
+		if !(test.skipOnWindows && runtime.GOOS == "windows") {
+			_, err := d.NodePublishVolume(context.Background(), &test.req)
+			if !testutil.AssertError(&test.expectedErr, err) {
+				t.Errorf("test case: %s, \nUnexpected error: %v\nExpected error: %v", test.desc, err, test.expectedErr.GetExpectedError())
+			}
 		}
 	}
 
 	// Clean up
-	err := os.RemoveAll(targetTest)
+	err = os.RemoveAll(targetTest)
 	assert.NoError(t, err)
 	err = os.RemoveAll(alreadyMountedTarget)
 	assert.NoError(t, err)
 }
 
 func TestNodeUnpublishVolume(t *testing.T) {
-	skipIfTestingOnWindows(t)
-	errorTarget := "./error_is_likely_target"
-	targetFile := "./abc.go"
-	targetTest := "./target_test"
+	errorTarget := testutil.GetWorkDirPath("error_is_likely_target", t)
+	targetFile := testutil.GetWorkDirPath("abc.go", t)
+	targetTest := testutil.GetWorkDirPath("target_test", t)
 
 	tests := []struct {
-		desc        string
-		req         csi.NodeUnpublishVolumeRequest
-		expectedErr error
+		desc          string
+		req           csi.NodeUnpublishVolumeRequest
+		expectedErr   testutil.TestError
+		skipOnWindows bool
 	}{
 		{
-			desc:        "[Error] Volume ID missing",
-			req:         csi.NodeUnpublishVolumeRequest{TargetPath: targetTest},
-			expectedErr: status.Error(codes.InvalidArgument, "Volume ID missing in request"),
+			desc: "[Error] Volume ID missing",
+			req:  csi.NodeUnpublishVolumeRequest{TargetPath: targetTest},
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.InvalidArgument, "Volume ID missing in request"),
+			},
 		},
 		{
-			desc:        "[Error] Target missing",
-			req:         csi.NodeUnpublishVolumeRequest{VolumeId: "vol_1"},
-			expectedErr: status.Error(codes.InvalidArgument, "Target path missing in request"),
+			desc: "[Error] Target missing",
+			req:  csi.NodeUnpublishVolumeRequest{VolumeId: "vol_1"},
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.InvalidArgument, "Target path missing in request"),
+			},
 		},
 		{
-			desc:        "[Error] Unmount error mocked by IsLikelyNotMountPoint",
-			req:         csi.NodeUnpublishVolumeRequest{TargetPath: errorTarget, VolumeId: "vol_1"},
-			expectedErr: status.Error(codes.Internal, "failed to unmount target \"./error_is_likely_target\": fake IsLikelyNotMountPoint: fake error"),
+			desc: "[Error] Unmount error mocked by IsLikelyNotMountPoint",
+			req:  csi.NodeUnpublishVolumeRequest{TargetPath: errorTarget, VolumeId: "vol_1"},
+			// todo: This test does not return any error on windows
+			// Once the issue is figured out, we'll remove this field
+			skipOnWindows: true,
+			expectedErr: testutil.TestError{
```

Comment:
In some cases, CSI proxy does not return error on Windows. Only these specific test cases are skipped on Windows for now, and the remaining test cases are executed. I think we may have to change the assertion logic a bit around here as mount behavior is different on Windows and Linux. I will open a follow-up PR to enable this test case.